### PR TITLE
feat(core-flows,types,pricing,medusa): Products API can create prices with rules

### PR DESCRIPTION
**Pull request by riqwan**
what:

- product & variants create/update apis accepts rules
- responses will now contain rules in prices
- cleanup price set rules

RESOLVES CORE-2356

### CODE REVIEW

File: packages/modules/pricing/src/services/pricing-module.ts [hunk 1]
```diff
@@ -744,56 +667,27 @@ export default class PricingModuleService
...
-        (d) => d.prices?.map((ma) => Object.keys(ma?.rules ?? {})).flat() ?? []
-      )
-      .flat()
-      .filter((r) => !ruleTypeMap.has(r))
-
-    if (invalidMoneyAmountRule.length > 0) {
-      throw new MedusaError(
-        MedusaError.Types.INVALID_DATA,
-        `Rule types don't exist for money amounts with rule attribute: ${invalidMoneyAmountRule.join(
+        `Rule types don't exist for money amounts with rule attribute: ${invalidRuleAttributes.join(
```
Comment (olivermrbl) [inline packages/modules/pricing/src/services/pricing-module.ts]:
```suggestion
        `Rule types don't exist for prices with rule attribute: ${invalidRuleAttributes.join(
```

**Review by olivermrbl — APPROVED** [review]
Nice, LGTM!

What prompted the clean-up? (maybe this is obvious, but just me missing some context)

**Comment by riqwan** [comment]
@olivermrbl just removing some stuff we don't use anymore while I was in the area. We don't use the price set rules anywhere anymore. These were some leftovers.

**Comment by riqwan** [comment]
@fPolic just FYI, this one is ready 🤞🏻

**Comment by fPolic** [comment]
> @fPolic just FYI, this one is ready 🤞🏻

🏅

**Pull request merged**